packages/flutter/test/widgets/pinned_header_sliver_test.dart

```diff
@@ -0,0 +1,179 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('PinnedHeaderSliver basics', (WidgetTester tester) async {
```

Comment:
This is vertical+not-reversed. There should also be a basic test for vertical+reverse, horizontal+not-reversed, and horizontal+reversed. These cases tend to come back to us in bug reports. :)

Reply:
I have added tests for the other cases.

Reply:
Nice!